iris-mpc-cpu/src/execution/hawk_main.rs

```diff
@@ -833,19 +847,43 @@ impl HawkResult {
     }
 
     fn job_result(self) -> ServerJobResult {
+        use Orientation::Mirror;
+        use StoreId::{Left, Right};
+
         let n_requests = self.is_matches.len();
 
         let matches = self.is_matches().to_vec();
+
         let match_ids = self.match_ids();
 
-        let partial_match_ids_left = self
-            .match_results
-            .filter_map(|(id, [l, _r])| l.then_some(id.index()));
-        let partial_match_ids_right = self
-            .match_results
-            .filter_map(|(id, [_l, r])| r.then_some(id.index()));
-        let partial_match_counters_left = partial_match_ids_left.iter().map(Vec::len).collect();
-        let partial_match_counters_right = partial_match_ids_right.iter().map(Vec::len).collect();
+        let (partial_match_ids_left, partial_match_counters_left) = self.select(Filter {
+            eyes: Only(Left),
+            orient: Both,
+        });
+
+        let (partial_match_ids_right, partial_match_counters_right) = self.select(Filter {
+            eyes: Only(Right),
+            orient: Both,
+        });
+
+        let (full_face_mirror_match_ids, _) = self.select(Filter {
+            eyes: Both,
+            orient: Only(Mirror),
+        });
+
+        let (full_face_mirror_partial_match_ids_left, full_face_mirror_partial_match_counters_left) =
+            self.select(Filter {
+                eyes: Only(Left),
```

Comment:
TODO: Not sure which left it is with regards to mirroring.

Reply:
In GPU, these refer to the DB sides, i.e., we swap the request sides for mirroring: LEFT => right query vs left DB.
I think this should be fine as is then?